### PR TITLE
Fixed the OPE_ID quotes getting stripped issue

### PR DESCRIPTION
**Pull request by maneshkommidi**

### CODE REVIEW

**Review by nagaborra — APPROVED** [review]

**Comment by saintsoup52** [comment]
Yay!!!

I am soo... soo... soo... excited about this!

**Review by patrickvinograd — COMMENTED** [review]
(Mis-submitted review, see below)

**Review by patrickvinograd — CHANGES_REQUESTED** [review]
This needs test cases to ensure that:
- The exported CSV has quotes preserved in the OPE_ID
- The data presented by the API is unaffected with quoted OPE_IDs.

**Comment by patrickvinograd** [comment]
For example: https://api.vets.gov/v0/gi/institutions/11919137 has 
`"ope":"00321600"`
string key, string value. Make sure with this change that value looks the same, is not double-quoted or anything.

**Comment by maneshkommidi** [comment]
@patrickvinograd added test cases and fixed the api response OPE field to have the exact string.

File: app/controllers/v0/institutions_controller.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def index
     # GET /v0/institutions/20005123
     def show
       resource = Institution.version(@version[:number]).find_by(facility_code: params[:id])
-
+      resource[:ope] = resource[:ope].tr('\"', '') if (resource.present? && resource[:ope].present?)
```
Comment (jkassemi) [inline app/controllers/v0/institutions_controller.rb]: Is this a candidate for inclusion in the model or possibly something that can be worked into a converter process?

Comment (maneshkommidi) [reply app/controllers/v0/institutions_controller.rb]: @jkassemi We cannot write this piece of code in to converter process,  as in CSV we need quotes for OPE_ID and coming to model I think it does not make big difference as we are querying the data here.

Comment (jkassemi) [reply app/controllers/v0/institutions_controller.rb]: What about the serializer?

Would like, where possible, to maintain focus in the controller on authentication, input sanitization, data lookup, and orchestrating rendering. This seems like presentation logic within the controller.

Comment (maneshkommidi) [reply app/controllers/v0/institutions_controller.rb]: Thanks, Updated the code

**Review by jkassemi — COMMENTED** [review]
Thanks, @maneshkommidi, looks great! One more thing, can you add some specs for the `InstitutionProfileSerializer`?

**Comment by maneshkommidi** [comment]
@jkassemi added spec's for  InstitutionProfileSerializer, when you get a chance please check and let me know your thoughts.

File: spec/serializers/institution_profile_serializer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    expect(data['id'].to_i).to eq(institution.id)
+  end
+
+  it 'links to vets api' do
+    expect(links['vet_website_link']).to_not be_empty
+  end
+
+  it 'should include OPE value' do
+    expect(data['attributes']['ope']).to eq(institution.ope)
+  end
```
Comment (jkassemi) [inline spec/serializers/institution_profile_serializer_spec.rb]: So from my reading on this the ope value (`object.ope` from `InstitutionProfileSerializer `) can be one of three things before it hits the serializer:

1) a non-nil value containing a "
2) a non-nil value that doesn't contain a "
3) a nil value

So to make sure this feature is working as intended, we'd need to ensure that this spec hits each of those variants.

**Comment by jkassemi** [comment]
@patrickvinograd mind taking another look?

**Comment by jkassemi** [comment]
@patrickvinograd bump

**Review by patrickvinograd — CHANGES_REQUESTED** [review]
I pulled this down locally and it does not appear to have the desired effect. I think it would work if OPEs were quoted in the input file (crosswalk.csv), but they are not. To be clear GIDS is handling the values correctly as strings all the way through upload, processing, cross-referencing, exposing via the API, it's just that when we *export* a CSV, we need to add quotes. 

While we could add quotes at import time, I believe this would be error prone in some of the processing/cross-correlation that is done among tables. Instead, I think the smallest change would be to just quote the ope value at export time. That could be done by introducing a "export_converter" field to the `CSV_CONVERTER_INFO` dict in the Institution model class, and using it in `lib/csv_helper/exporter.rb`. The vast majority of columns will have no export_converter so should pass through as-is, but the ope column of the Institution model will get a converter that quotes the values.

**Comment by patrickvinograd** [comment]
Above suggested change should result in no impact to the API data or to how the institution table is built and stored, and only changes how it is exported as a CSV which is the desired effect.

**Comment by saintsoup52** [comment]
The "input file" should have quotes around the OPE IDs... and they will in the future.

The problem is that the GIDS has been stripping them for so long that the files we are inputting now don't have the quotes. I can provide a new version with the quotes for testing.

**Comment by jkassemi** [comment]
I reread the original issue https://github.com/department-of-veterans-affairs/gibct-data-service/issues/304 - I'd thought this was a fix for a different problem. There's no issue linked here, so assuming that's the one, and I'm closing. This is not the right fix for the problem.

**Pull request closed without merging**